apps/prairielearn/src/pages/administratorQuery/administratorQuery.html.ts

```diff
@@ -0,0 +1,353 @@
+});
+type AdministratorQuery = z.infer<typeof AdministratorQuerySchema>;
+
+export const AdministratorQueryQueryRunSchema = z.object({
+  formatted_date: z.string().optional(),
+  sql: z.string(),
+  params: z.record(z.any()),
+  error: z.string().nullable(),
+  result: AdministratorQueryResultSchema,
+  authn_user_id: z.string().optional(),
+  name: z.string().optional(),
+  id: z.string().optional(),
+});
+export type AdministratorQueryQueryRun = z.infer<typeof AdministratorQueryQueryRunSchema>;
+
+export function AdministratorQuery({
+  resLocals,
+  has_query_run,
+  query_run_id,
+  formatted_date,
+  params,
+  error,
+  result,
+  sqlFilename,
+  info,
+  sqlHighlighted,
+}: {
+  resLocals: Record<string, any>;
+  has_query_run: boolean;
+  query_run_id: string | null;
+  formatted_date: string | undefined;
+  params: AdministratorQueryParamsRecord | undefined;
+  error: string | null;
+  result: AdministratorQueryResult | null;
+  sqlFilename: string;
+  info: AdministratorQuery;
+  sqlHighlighted: string;
+}) {
+  function renderHeader(columns, col) {
+    const row = {};
+    columns.forEach((c) => {
+      row[c] = true;
+    });
+
+    if (col === 'course_id' && 'course' in row) {
+      return '';
+    } else if (col === 'course_instance_id' && 'course_instance' in row) {
+      return '';
+    } else if (col === 'assessment_id' && 'assessment' in row && 'course_instance_id' in row) {
+      return '';
+    } else if (/^_sortval_/.test(col)) {
+      return '';
+    } else {
+      return html`<th>${col}</th>`;
+    }
+  }
+
+  function render(row, col) {
+    const tdAttributes =
+      `_sortval_${col}` in row ? html`data-text="${row['_sortval_' + col]}"` : html``;
+
+    if (col === 'course_id' && 'course' in row) {
+      null;
+    } else if (col === 'course' && 'course_id' in row) {
+      return html`
+        <td ${tdAttributes}>
+          <a href="${`${resLocals.urlPrefix}/course/${row['course_id']}`}">${row[col]}</a>
+        </td>
+      `;
+    } else if (col === 'course_instance_id' && 'course_instance' in row) {
+      null;
+    } else if (col === 'course_instance' && 'course_instance_id' in row) {
+      return html`
+        <td ${tdAttributes}>
+          <a href="${`${resLocals.urlPrefix}/course_instance/${row['course_instance_id']}`}"
+            >${row[col]}</a
+          >
+        </td>
+      `;
+    } else if (col === 'assessment_id' && 'assessment' in row && 'course_instance_id' in row) {
+      null;
+    } else if (col === 'assessment' && 'assessment_id' in row && 'course_instance_id' in row) {
+      return html`
+        <td ${tdAttributes}>
+          <a
+            href="${`${resLocals.urlPrefix}/course_instance/${row['course_instance_id']}/instructor/assessment/${row['assessment_id']}`}"
+            >${row[col]}</a
+          >
+        </td>
+      `;
+    } else if (/^_sortval_/.test(col)) {
+      null;
+    } else if (row[col] == null) {
+      return html` <td ${tdAttributes}></td> `;
```

Comment:
Prettier won't get rid of these spaces for you. Double check that the rendered HTML looks fine without them, but it should. You should make this change elsewhere in this function too.

```suggestion
      return html`<td ${tdAttributes}></td>`;
```

Reply:
Spaces removed in commit 53b8ebc.